docs/overlays.md

```diff
@@ -0,0 +1,15 @@
+# Augmenting Auto-Generated Code w/ Overlays (Legacy Feature)
+
+> **Warning** Overlays are a legacy feature and support may be removed in a future version of the bridge.
+>             We do not recommend using overlays for a new provider. To augment a terraform provider,
+>             please use [tfbridge.ProviderInfo.MuxWith](./muxwith.md).
+
+An overlay is a set of additional directives that the code generator obeys when creating the final packages.
```

Comment:
Would it be helpful to call out that this is done per target programming language (which is also why it's deprecated)?